src/App.tsx

```diff
@@ -8,52 +8,64 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
 import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { fetchUsers } from './features/users/usersSlice';
+import { setAuthor } from './features/author/authorSlice';
+import { fetchUserPosts, setPosts } from './features/posts/postsSlice';
+import { Post } from './types/Post';
+import { setSelectedPost as setSelectedPostAction }
+  from './features/selectedPost/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { author } = useAppSelector((store) => store.author);
+  const { value: selectedPost } = useAppSelector((store) => store.selectedPost);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { items: posts, loaded, hasError }
+    = useAppSelector((store) => store.posts);
```

Comment:
consider creating another hook usePosts for example it can look like 
`const usePosts = () => useAppSelector(store => store.posts)
}`
```suggestion
    = usePosts();
```